### ISSUE TITLE
performance tests between lua-resty-template and others

### ISSUE THREAD

**Issue by sparvu** [bungle/lua-resty-template#3]
Hi,

We are looking into selecting a template engine for our coming LUA applications. We been now reading between your module and etlua. If I understood right both are doing the same.

Is it possible for future to consider adding some comparative test results for resty-template vs etlua or others ... or a section for README regarding performance consideration regarding resty-template ?

Thanks

**Comment by splitice** [comment]
I would be interested in this too.

I did some testing with some basic SSI templates I converted to lua-resty-template and saw maybe a 15% difference (hard to tell, not consistent or all that large, possibly GC?) with the edge going to SSI.

I was surprised by that given the lack of caching in SSI, I guess theres also overheads for using a higher level language like lua.

Still I think its fast enough for us.

**Comment by bungle** [comment]
@splitice, Can you provide your example, so that I can try it out as well?

**Comment by bungle** [comment]
@splitice you could also benchmark with this patch on LuaJIT (not sure if it makes difference):
https://github.com/LuaJIT/LuaJIT/pull/169

Discussion:
https://github.com/LuaJIT/LuaJIT/issues/168
https://github.com/LuaJIT/LuaJIT/issues/171

Regards
Aapo

**Comment by splitice** [comment]
Sorry, I cant provide the exact template since I converted our templates we actually use from SSI -> lua-resty-template.

The main template tested isnt overly big and is just three variable substitutions into a HTML page. Theres a good chance that PR might decrease the difference. I +1'ed that PR earlier. Looks promising for some of the other stuff we do.

I was a bit surprised that it didnt beat SSI given that the lua bytecode is cached, whereas I believe SSI is parsed each time. It might be too simple of a test to really bring that out I suppose, the overhead of starting the Lua engine may be greater than the overhead of parsing SSI tags.

I tested using apache-bench over localhost so it includes the full processing time of the http server.

If I find some time I'll see what I can do about producing a more reproducible test. I'll also upgrade LuaJIT (we are still on 2.1beta1) and the Lua module.
